projects/js-packages/publicize-components/src/hooks/use-anaytics-tracks/index.js

```diff
@@ -0,0 +1,90 @@
+import jetpackAnalytics from '@automattic/jetpack-analytics';
+import { useConnection } from '@automattic/jetpack-connection';
+import { useCallback, useEffect } from 'react';
+
+const useAnalyticsTracks = ( {
+	pageViewEventName,
+	pageViewNamespace = 'jetpack',
+	pageViewSuffix = 'page_view',
+	pageViewEventProperties = {},
+} = {} ) => {
+	const { isUserConnected, isRegistered, userConnectionData } = useConnection();
+	const { login, ID } = userConnectionData.currentUser?.wpcomUser || {};
+	const { tracks } = jetpackAnalytics;
+	const { recordEvent } = tracks;
+
+	/**
+	 * Record an event if the user has a Jetpack connection
+	 *
+	 * @type {(function(*, *): void)|*}
+	 */
+	const recordEventIfConnectedUser = useCallback(
+		( eventName, properties ) => {
+			// Do nothing if there is not a connected user.
+			if ( ! ( isUserConnected && ID && login ) ) {
+				return;
+			}
+
+			recordEvent( eventName, properties );
+		},
+		[ recordEvent, isUserConnected, ID, login ]
+	);
+
+	const recordEventAsync = useCallback(
+		async ( event, properties ) => {
+			recordEventIfConnectedUser( event, properties );
+		},
+		[ recordEventIfConnectedUser ]
+	);
+
+	const recordEventHandler = useCallback(
+		( eventName, properties, callback = () => {} ) => {
+			/*
+			 * `properties` is optional,
+			 * meaning it can be actually the `callback`.
+			 */
+			callback = typeof properties === 'function' ? properties : callback;
+			properties = typeof properties === 'function' ? {} : properties;
+
+			return () => recordEventAsync( eventName, properties ).then( callback );
+		},
+		[ recordEventAsync ]
+	);
+
+	// Initialize Analytics identifying the user.
+	useEffect( () => {
+		if ( ! ( isUserConnected && ID && login ) ) {
+			return;
+		}
+
+		jetpackAnalytics.initialize( ID, login );
+	}, [ isUserConnected, ID, login ] );
+
+	/*
+	 * Track page-view event.
+	 * It's considered a page view event
+	 * when the component is mounted.
+	 */
+	const pageViewEvent = pageViewEventName
+		? `${ pageViewNamespace }_${ pageViewEventName }_${ pageViewSuffix }`
+		: null;
+
+	useEffect( () => {
+		// Also, only run if the site is registered.
+		if ( ! isRegistered ) {
+			return;
+		}
+
+		if ( ! pageViewEvent ) {
+			return;
+		}
+
+		recordEventIfConnectedUser( pageViewEvent, pageViewEventProperties );
+	}, [] ); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
You shouldn't need to use this eslint disable rule. What is the motivation behind this?

Reply:
I understand it's to limit the execution of the hook to the one time. However, can we be sure that the final value of `isRegistered` is fetch on the first render? If you want to achieve similar effect, we can use the natural indicators (build from props passed to the hook) or artificial one (a flag)

Reply:
I grabbed this file from another plugin, so I did not search too deeply for improvements 😅 . Thanks for pointing this out!